tests/integration/webclient/src/main/java/io/helidon/tests/integration/webclient/GreetService.java

```diff
@@ -201,6 +205,24 @@ private void updateGreetingHandler(ServerRequest request,
                 .exceptionally(ex -> processErrors(ex, request, response));
     }
 
+    private void connectionClose(ServerRequest request, ServerResponse response) {
+        response.send(Multi
+                .interval(100, 500, TimeUnit.MILLISECONDS,
+                        Executors.newSingleThreadScheduledExecutor())
+                .map(i -> "item" + i)
+                .limit(15)
+                .peek(s -> {
+                    if ("item2".equals(s)) {
+                        Main.webServer.shutdown();        // abrupt shutdown to detect connection close
```

Comment:
Test can be made little more elegant with `ServerRequest#closeConnection` #3380

Reply:
I think I like the abruptness of the shutdown to simulate a server going down unexpectedly. I will try the close too anyway.